### PR TITLE
[IMP] account_edi,account_edi_facturx: Add generic test suite + factu…

### PR DESCRIPTION
**Pull request by smetl**
…rx tests

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account_edi_facturx/tests/test_facturx.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        '''
+
+    def test_facturx(self):
+        ''' Test the generated Facturx Edi attachment without any modification of the invoice. '''
+
+        with self.mocked_today('2017-02-01'):
+            self.invoice.post()
+
+            facturx_attachment = self.invoice.edi_document_ids
+            self.assertAttachmentXmlEqual(facturx_attachment, self.expected_invoice_facturx_values)
```
Comment (qdp-odoo) [inline addons/account_edi_facturx/tests/test_facturx.py:159]:
```suggestion
            self.assertXmlTreeEqual(self.extract_xml(facturx_attachment), self.expected_invoice_facturx_values)
```

File: addons/account_edi_facturx/tests/test_facturx.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        })
+
+        with self.mocked_today('2017-02-01'):
+            self.invoice.post()
+
+            facturx_attachment = self.invoice.edi_document_ids
+            self.assertAttachmentXmlEqual(
+                facturx_attachment,
+                self.expected_invoice_facturx_values,
+                diff_expected_xml_str='''
```
Comment (qdp-odoo) [inline addons/account_edi_facturx/tests/test_facturx.py:174]:
```diff
+           diff_expected_xml_str=''' bablabalabla '''
-            self.assertAttachmentXmlEqual(
-                facturx_attachment,
-                self.expected_invoice_facturx_values,
-                diff_expected_xml_str='''
+            self.assertXmlTreeEqual(
+                self.extract_xml(facturx_attachment),
+                self.applies_xpath(self.expected_invoice_facturx_values, diff_expected_xml_str),
```

**Review by qdp-odoo — CHANGES_REQUESTED** [review]
what about something like that?

**Comment by smetl** [comment]
@qdp-odoo Changes made as requested

**Comment by smetl** [comment]
Merged in 13.0. See: https://github.com/odoo/odoo/pull/53895

**Comment by qdp-odoo** [comment]
fix landed in 12.0 with https://github.com/odoo/odoo/pull/53894 and was manually forward ported in 13.0, with the test suite, with https://github.com/odoo/odoo/pull/53895

**Pull request closed without merging**